platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -785,6 +791,77 @@ def get_tx_fault(self):
         api = self.get_xcvr_api()
         return [False] * api.NUM_CHANNELS if api else None
 
+    def get_temperature(self):
+        try:
+            if not self.is_sw_control():
```

Comment:
@Junchao-Mellanox can we make use of TRANSCIEVER_INFO table to decide sw or fw control?